From baef22e0549112acbaefa09d263acc8c3f1478bb Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <b02247@freescale.com>
Date: Tue, 5 Nov 2013 13:45:29 +0800
Subject: [PATCH 42/72] ENGR00285689 Audio playback speed is slower for some
 format.

For 96KHz samplerate bitstream, the playback speed is slower than
normal speed. Audioflinger default only support 2x samplerate converter.
So enlarge it to 8x.

Signed-off-by: Shengjiu Wang <b02247@freescale.com>
---
 services/audioflinger/Threads.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 5d100b6..268149b 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -3171,8 +3171,8 @@ AudioFlinger::PlaybackThread::mixer_state AudioFlinger::MixerThread::prepareTrac
                 name,
                 AudioMixer::TRACK,
                 AudioMixer::CHANNEL_MASK, (void *)track->channelMask());
-            // limit track sample rate to 2 x output sample rate, which changes at re-configuration
-            uint32_t maxSampleRate = mSampleRate * 2;
+            // limit track sample rate to 8 x output sample rate, which changes at re-configuration
+            uint32_t maxSampleRate = mSampleRate * 8;
             uint32_t reqSampleRate = track->mAudioTrackServerProxy->getSampleRate();
             if (reqSampleRate == 0) {
                 reqSampleRate = mSampleRate;
-- 
1.8.0

